packages/builder/src/pages/builder/_layout.svelte

```diff
@@ -113,66 +250,39 @@
       })
     }
 
-    // lastly
     await analyticsPing()
+  }
+
+  onMount(() => {
+    initPromise = initBuilder()
+    hasAuthenticated = !!$auth.user
   })
 
+  // Re-run initBuilder when user logs in
   $: {
-    const apiReady = $admin.loaded && $auth.loaded
-
-    // firstly, set the return url
-    if (
-      loaded &&
-      apiReady &&
-      !$auth.user &&
-      !CookieUtils.getCookie(Constants.Cookies.ReturnUrl) &&
-      // logout triggers a page refresh, so we don't want to set the return url
-      !$auth.postLogout &&
-      // don't set the return url on pre-login pages
-      !$isActive("./auth") &&
-      !$isActive("./invite") &&
-      !$isActive("./admin")
-    ) {
-      const url = window.location.pathname
-      CookieUtils.setCookie(Constants.Cookies.ReturnUrl, url)
+    const isAuthenticated = !!$auth.user
+    if (isAuthenticated && !hasAuthenticated) {
```

Comment:
These variable names are very confusing to me. `isAuthenticated` versus `hasAuthenticated`. What does that mean?